GoogleDataTransport/GDTCORLibrary/Internal/GDTCORDirectorySizeCalculator.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import "GoogleDataTransport/GDTCORLibrary/Public/GoogleDataTransport/GDTCORStorageProtocol.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** The class calculates and caches the specified directory content size and uses add/remove signals
```

Comment:
There are no thread-safety or locking mechanisms of any kind in this class. Can you speak to why that is and how to use this class in the doc?